packages/oid4vci-issuer-rest-api/oid4vci-openapi.yml

```diff
@@ -0,0 +1,653 @@
+          description: Background color for display components
+        text_color:
+          type: string
+          description: Text color for display components
+
+    CredentialsSupportedDisplay:
+      type: object
+      description: Display information for a credential type
+      properties:
+        name:
+          type: string
+          description: Name of the credential for display purposes
+        locale:
+          type: string
+          description: Language locale for the credential display
+        logo:
+          type: object
+          properties:
+            url:
+              type: string
+              description: URL to the credential logo
+            alt_text:
+              type: string
+              description: Alternative text for the logo
+        background_color:
+          type: string
+          description: Background color for the credential display
+        text_color:
+          type: string
+          description: Text color for the credential display
+
+paths:
+  /.well-known/openid-credential-issuer:
+    get:
+      summary: Get credential issuer metadata
+      description: Returns the metadata about the credential issuer's capabilities
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IssuerMetadata'
+
+  /.well-known/oauth-authorization-server:
+    get:
+      summary: Get authorization server metadata
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                type: object
+
+  /webapp/credential-offer-status:
+    post:
+      summary: Get credential issuance status
+      description: Check the current status of a credential issuance request
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              type: object
+              required:
+                - id
+              properties:
+                id:
+                  type: string
+                  description: Identifier of the credential offer
+                  example: "b326c4ab-16f1-4304-83e1-4637f8fe4207"
+      responses:
+        '200':
+          description: Successful response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/IssueStatusResponse'
+        '404':
+          description: Credential offer not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+
+  /authorization-challenge:
+    post:
+      summary: Authorization challenge endpoint
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/AuthorizationChallengeRequest'
+      responses:
```

Comment:
can result in a 400, but looking at the code we should probably improve that a bit, as we should also throw a 500 if something goes wrong...